lib/schema/validate.js

```diff
@@ -6,6 +6,49 @@ function loadSchema(schemaPath) {
 	return JSON.parse(readFileSync(path.join(__dirname, schemaPath), {encoding: "utf8"}));
 }
 
+class ValidationError extends Error {
+	constructor(errors) {
+		super();
+
+		this.name = "ValidationError";
+
+		this.errors = ValidationError.filterErrors(errors)
+			.map(ValidationError.formatError);
+
+		this.message = ValidationError.formatMessage(this.errors);
+
+		Error.captureStackTrace(this, this.constructor);
+	}
+
+	static formatMessage(errors) {
+		return errors.map((err) => err.message).join("\n");
+	}
+
+	static filterErrors(allErrors) {
+		return allErrors.filter((error, i, errors) => {
+			if (error.keyword === "if") {
+				return false;
+			}
+
+			if (errors.findIndex(($) => {
+				return $.dataPath === error.dataPath &&
```

Comment:
This is wild 🦁, can we make this an `if` and maybe add another variable 🙂?

Reply:
Done